scripts/debug_tools/prepare_all_cmd_for_ctu.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python
+# -------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -------------------------------------------------------------------------
+import argparse
+import json
+import os
+import subprocess
+import sys
+
+import prepare_compile_cmd
+import prepare_compiler_includes
+import prepare_compiler_target
+import prepare_analyzer_cmd
+
+
+def execute(cmd):
+    print("Executing command: " + ' '.join(cmd))
+    try:
+        proc = subprocess.Popen(cmd,
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE)
+        out, err = proc.communicate()
+
+        print("stdout:\n\n" + out.decode("utf-8"))
+        print("stderr:\n\n" + err.decode("utf-8"))
+
+        if proc.returncode != 0:
+            print('Unsuccessful run: "' + ' '.join(cmd) + '"')
+            raise Exception("Unsuccessful run of command.")
+        return out
+    except OSError:
+        print('Failed to run: "' + ' '.join(cmd) + '"')
+        raise
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description='Prepare all commands '
+        'to execute in local environmennt for debugging.')
+    parser.add_argument('--clang', required=True)
+    parser.add_argument('--clang_plugin_name', default=None)
+    parser.add_argument('--clang_plugin_path', default=None)
+    args = parser.parse_args()
+
+    if "reports/failed" not in str(os.getcwd()) or\
+        os.path.basename(os.path.normpath(os.getcwd())) !=\
+            "failed":
+        print("Please cd to .../reports/failed")
+        sys.exit(-1)
+
+    tools_dir = os.path.dirname(os.path.realpath(__file__))
+
+    sources_root = "./sources-root"
+
+    compile_cmd_debug = "compile_cmd_DEBUG.json"
+    with open(compile_cmd_debug, 'w') as f:
+        f.write(
+            json.dumps(
+                prepare_compile_cmd.prepare(
+                    "../compile_cmd.json",
+                    sources_root),
+                indent=4))
+
+    compiler_includes_debug = "compiler_includes_DEBUG.json"
+    with open(compiler_includes_debug, 'w') as f:
```

Comment:
We have a JSON read helper function, but many parts of this code is duplicated here. You could try passing the `prepare_XYZ` as an argument to a wrapper? What are these types?